astropy/utils/tests/test_data.py

```diff
@@ -48,13 +45,29 @@ def test_download_nocache():
 
 @pytest.mark.remote_data('astropy')
 def test_download_parallel():
-    from ..data import download_files_in_parallel
-
-    fnout = download_files_in_parallel(
-        ['http://data.astropy.org',
-         'https://astropy.stsci.edu/data/intersphinx/README'])
+    import shelve
+    from ..data import (download_file, download_files_in_parallel,
+                        _get_download_cache_locs, get_cached_urls)
+
+    main_url = 'http://data.astropy.org/intersphinx/README'
+    mirror_url = 'http://www.astropy.org/astropy-data/intersphinx/README'
+    fnout = download_files_in_parallel([
+        'http://data.astropy.org', main_url, mirror_url])
     assert all([os.path.isfile(f) for f in fnout]), fnout
 
+    # Now test that download_file looks in mirror's cache before download.
+    # https://github.com/astropy/astropy/issues/6982
+    dldir, urlmapfn = _get_download_cache_locs()
+    with shelve.open(urlmapfn) as url2hash:
```

Comment:
NOTE: Use ` _open_shelve` for PY2 compat (see example in 2.x code in `data.py`).